app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt

```diff
@@ -0,0 +1,178 @@
+package org.mifos.mobile.viewModels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import org.mifos.mobile.api.BaseApiManager
+import org.mifos.mobile.api.DataManager
+import org.mifos.mobile.api.local.PreferencesHelper
+import org.mifos.mobile.models.Page
+import org.mifos.mobile.models.User
+import org.mifos.mobile.models.client.Client
+import org.mifos.mobile.models.payload.LoginPayload
+import org.mifos.mobile.utils.Constants
+import retrofit2.HttpException
+import java.net.UnknownHostException
+import javax.inject.Inject
+
+class LoginViewModel @Inject constructor(private val dataManager: DataManager?) : ViewModel() {
+    private val preferencesHelper: PreferencesHelper? = dataManager?.preferencesHelper
+    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
+    var error = MutableLiveData<String>()
+    var success = MutableLiveData<String>()
+    val navigateToPassCodeActivity = MutableLiveData<String>()
+    val userName = MutableLiveData<String>()
+    val Password = MutableLiveData<String>()
+
+
+    fun login(loginPayload: LoginPayload?) {
+        if (isCredentialsValid(loginPayload)) {
+            compositeDisposable.add(
+                dataManager?.login(loginPayload)
```

Comment:
We were supposed to use Coroutines, right?

Reply:
yes jawid , but during the review of sprint 1 , rajan asked to continue with RxJava for the time being